module/VuFind/src/VuFind/Db/Table/LoginToken.php

```diff
@@ -109,31 +110,42 @@ public function saveToken(
      */
     public function matchToken(array $token): ?LoginTokenRow
     {
-        $row = $this->getBySeries($token['series'], $token['user_id']);
-        if ($row && hash_equals($row['token'], hash('sha256', $token['token']))) {
-            if (time() > $row['expires']) {
-                $row->delete();
-                return null;
+        $rowsFound = false;
+        foreach ($this->select(['user_id' => $token['user_id'], 'series' => $token['series']]) as $row) {
```

Comment:
I see that you factored this to stop using the getBySeries method (and I understand why). Does this mean that getBySeries is now useless and should be removed? Or should we keep the method and add an argument to switch between returning "current" and "all" so we can still use it here?